python_modules/dagster/dagster/_core/execution/plan/execute_step.py

```diff
@@ -103,8 +114,14 @@ def _process_user_event(
             yield from _process_user_event(step_context, check_result)
 
         with disable_dagster_warnings():
+            if isinstance(user_event, MaterializeResult):
+                value = user_event.passed_value
+                value_unset = user_event.value_unset
+            else:
+                value, value_unset = None, True
```

Comment:
At minimum this is comment worthy as I would expect value to be a sentinel here as well

Reply:
made it so that this is no longer necessart